File: homework7/AssignmentCode/GymAppData/Models/Workout.cs
namespace GymAppData.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    public class Workout
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int WorkoutId { get; set; }

        public string OwnerId { get; set; }

        [Required]
        [StringLength(500)]
        public string Name { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime Created { get; set; }

        public bool IsDeleted { get; set; }

        public DateTime? Updated { get; set; }

        public virtual List<Day> Days { get; set; }
    }
}

File: homework7/AssignmentCode/GymAppCore/ClientApp/src/app/classes/Workout.ts
import { Day } from './Day';

export class Workout {
  name: string;
  workoutId: number;
  days: Day[];
  isDeleted: boolean;

  constructor(name: string) {
    this.name = name;
    this.workoutId = 0;
    this.days = [];
  }
}
